pandas/_libs/lib.pyx

```diff
@@ -724,9 +734,12 @@ cpdef ndarray[object] ensure_string_array(
             continue
 
         if not checknull(val):
-            result[i] = str(val)
+            if coerce =="all" or coerce == "non-null":
```

Comment:
u need to validate the coerce arg (and add a test that it raises properly if bad values are passed)